src/storage/accessors/InMemoryDataAccessor.ts

```diff
@@ -0,0 +1,170 @@
+import { Readable } from 'stream';
+import arrayifyStream from 'arrayify-stream';
+import { DataFactory } from 'n3';
+import type { NamedNode } from 'rdf-js';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import type { MetadataController } from '../../util/MetadataController';
+import { ensureTrailingSlash } from '../../util/Util';
+import type { DataAccessor } from './DataAccessor';
+
+interface DataEntry {
+  data: any[];
+  metadata: RepresentationMetadata;
+}
+interface ContainerEntry {
+  entries: { [name: string]: CacheEntry };
+  metadata: RepresentationMetadata;
+}
+type CacheEntry = DataEntry | ContainerEntry;
+
+class ArrayReadable extends Readable {
```

Comment:
Probably linter, but I tend to put the main class on top, and auxiliary classes below. As you wish.

Reply:
Not sure what the linter would say about it, I just have the habit of doing the opposite 😄 .